design/protected-branches.md

```diff
@@ -0,0 +1,38 @@
+# Protected Branches
+
+## Requirements
+
+1. As a lakeFS user, I should be able to mark branches as "protected" - from the UI, CLI and API.
```

Comment:
My thoughts about "branch protection":
Branch protection is a repository-level set of rules that we check for specific operations.
The user can define a rule that matches zero to many branches, patterns like: "dev-*" that apply a specific set of constraints.
One of the constraints is 'merge only' - which allows only merge operations on the branch.
As repository wide operations we should keep them at the same level as we keep retention (think it is option two here).